### PR TITLE
Feature/upgrade go sdk v11

### PR DESCRIPTION
**Pull request by ag-ramachandran**
**Description:** 
- Upgrade SDK for azure-kusto-go. Fixes issues for azblob updates
- Additional end to end tests. With the right env vars set, E2E tests can be performed on the exporter
- Update to Beta (stability) , since the API's have matured

**Link to tracking Issue:** 

#18886 
#18637 

**Testing:** 
image

End to end tests done

**Documentation:** 
- Upgrade SDK for azure-kusto-go
- Additional end to end tests
- Update to Beta (stability)

### CODE REVIEW

File: exporter/azuredataexplorerexporter/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 | Status                   |                       |
 | ------------------------ | --------------------- |
-| Stability                | [alpha]               |
+| Stability                | [beta]               |
```
Comment (mx-psi) [inline exporter/azuredataexplorerexporter/README.md:5]: This needs another changelog entry announcing the promotion to beta

**Review by mx-psi — COMMENTED** [review]
There are some merge conflicts, can you fix them with a merge/rebase against main?

**Comment by ag-ramachandran** [comment]
Sure @mx-psi, on it

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
Looks like you need to run `make gotidy`

**Comment by ag-ramachandran** [comment]
> Looks like you need to run `make gotidy`

Acked, missed that. Running that workflow. Sorry

**Comment by ag-ramachandran** [comment]
Hello @mx-psi , Sorry to disturb. Upgrades to this lib will need to go along with changes in azblob. How should we go about it.
```
Error: ../../receiver/azureblobreceiver/blobclient.go:30:24: undefined: azblob.ServiceClient
Error: ../../receiver/azureblobreceiver/blobclient.go:36:87: undefined: azblob.BlockBlobClient
Error: ../../receiver/azureblobreceiver/blobclient.go:66:31: undefined: azblob.NewServiceClientFromConnectionString
```

File: .chloggen/adx-update-go-lib.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
```
Comment (codeboten) [inline .chloggen/adx-update-go-lib.yaml:2]: if the change is just an update to remove a deprecated library, it doesn't need a changelog entry

**Comment by codeboten** [comment]
> Hello @mx-psi , Sorry to disturb. Upgrades to this lib will need to go along with changes in azblob. How should we go about it. [Trace](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4305280869/jobs/7507612480)
> 
> ```
> Error: ../../receiver/azureblobreceiver/blobclient.go:30:24: undefined: azblob.ServiceClient
> Error: ../../receiver/azureblobreceiver/blobclient.go:36:87: undefined: azblob.BlockBlobClient
> Error: ../../receiver/azureblobreceiver/blobclient.go:66:31: undefined: azblob.NewServiceClientFromConnectionString
> ```

I'd update both components in this pr if the dependency breaks it

File: .chloggen/adx-update-go-lib.yaml (hunk 2 above)
Comment (asaharn) [reply .chloggen/adx-update-go-lib.yaml:2]: Thanks. Will remove it.

**Review by codeboten — COMMENTED** [review]
@ag-ramachandran please resolve the conflicts

**Comment by ag-ramachandran** [comment]
> @ag-ramachandran please resolve the conflicts

Hello Alex @codeboten, 
Rebased and fixed the conflicts

Failed with the following in CI though. Please let us know if there is anything we can do help on this.


`make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/oteltestbedcol'
level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package azureblobreceiver: could not load export data: no export data for \"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureblobreceiver\""
level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package azureblobreceiver: could not load export data: no export data for \"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureblobreceiver\"\n\n"
make[2]: *** [../../Makefile.Common:166: lint] Error 3
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/configschema'
make[1]: *** [Makefile:178: cmd/configschema] Error 2
make[1]: *** Waiting for unfinished jobs....`

cc: @asaharn

**Comment by ag-ramachandran** [comment]
Hello @codeboten , @mx-psi .Fixed receiver code for azureblobreceiver. Will be great if you can review it too

File: cmd/configschema/go.mod
Comment (mx-psi) [inline cmd/configschema/go.mod:1]: Where the changes on this file intentional? I think we should revert them (this is making 'checks' fail and it sounds like some command went wrong)

**Review by mx-psi — COMMENTED** [review]
Looks like the CI 'checks' job is failing because of changes on configschema. I think these were probably unintentional, could you take a look?

File: cmd/configschema/go.mod
Comment (asaharn) [reply cmd/configschema/go.mod:1]: Hi @mx-psi , these are result go mod tidy.

**Comment by asaharn** [comment]
Hi @mx-psi , ci failing due to Test failure in `translator/loki`

image

**Comment by TylerHelmuth** [comment]
@asaharn that test is unrelated.  This PR should take care of the issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20241

**Review by codeboten — COMMENTED** [review]
Needs a `make gotidy`

**Comment by asaharn** [comment]
HI @codeboten @mx-psi,  
Done with all the changes to make CI happy.

**Pull request merged**